src/ctl.c

```diff
@@ -3664,6 +3678,10 @@ CTL_RO_CGEN(config_stats, stats_arenas_i_base,
 CTL_RO_CGEN(config_stats, stats_arenas_i_internal,
     atomic_load_zu(&arenas_i(mib[2])->astats->astats.internal, ATOMIC_RELAXED),
     size_t)
+CTL_RO_CGEN(config_stats, stats_arenas_i_metadata_edata,
+    arenas_i(mib[2])->astats->astats.metadata_edata, size_t)
+CTL_RO_CGEN(config_stats, stats_arenas_i_metadata_rtree,
+    arenas_i(mib[2])->astats->astats.metadata_rtree, size_t)
```

Comment:
The last step we need, is to output the new counters in stats here:
https://github.com/jemalloc/jemalloc/blob/005f20aa7fdef1be6f9fe46e4f2e7b88177a9f21/src/stats.c#L1353

After adding that, can you share the output from malloc_stats by running some test, e.g. `MALLOC_CONF=stats_print:true ./test/unit/a0`

Reply:
___ Begin jemalloc statistics ___
Version: "5.3.0-136-g2f99efb6ae7bc5532428723e167538ebd0409e0c"
Build-time option settings
  config.cache_oblivious: true
  config.debug: false
  config.fill: true
  config.lazy_lock: false
  config.malloc_conf: ""
  config.opt_safety_checks: false
  config.prof: false
  config.prof_libgcc: false
  config.prof_libunwind: false
  config.stats: true
  config.utrace: false
  config.xmalloc: false
Run-time option settings
  opt.abort: false
  opt.abort_conf: false
  opt.cache_oblivious: true
  opt.confirm_conf: false
  opt.retain: true
  opt.dss: "secondary"
  opt.narenas: 288
  opt.percpu_arena: "disabled"
  opt.oversize_threshold: 8388608
  opt.hpa: false
  opt.hpa_slab_max_alloc: 65536
  opt.hpa_hugification_threshold: 1992294
  opt.hpa_hugify_delay_ms: 10000
  opt.hpa_min_purge_interval_ms: 5000
  opt.hpa_dirty_mult: "0.25"
  opt.hpa_sec_nshards: 4
  opt.hpa_sec_max_alloc: 32768
  opt.hpa_sec_max_bytes: 262144
  opt.hpa_sec_bytes_after_flush: 131072
  opt.hpa_sec_batch_fill_extra: 0
  opt.metadata_thp: "disabled"
  opt.mutex_max_spin: 600
  opt.background_thread: false (background_thread: false)
  opt.dirty_decay_ms: 10000 (arenas.dirty_decay_ms: 10000)
  opt.muzzy_decay_ms: 0 (arenas.muzzy_decay_ms: 0)
  opt.lg_extent_max_active_fit: 6
  opt.junk: "false"
  opt.zero: false
  opt.experimental_infallible_new: false
  opt.tcache: true
  opt.tcache_max: 32768
  opt.tcache_nslots_small_min: 20
  opt.tcache_nslots_small_max: 200
  opt.tcache_nslots_large: 20
  opt.lg_tcache_nslots_mul: 1
  opt.tcache_gc_incr_bytes: 65536
  opt.tcache_gc_delay_bytes: 0
  opt.lg_tcache_flush_small_div: 1
  opt.lg_tcache_flush_large_div: 1
  opt.debug_double_free_max_scan: 0
  opt.thp: "default"
  opt.stats_print: true
  opt.stats_print_opts: ""
  opt.stats_print: true
  opt.stats_print_opts: ""
  opt.stats_interval: -1
  opt.stats_interval_opts: ""
  opt.zero_realloc: "free"
Arenas: 289
Quantum size: 16
Page size: 4096
Maximum thread-cached size class: 32768
Number of bin size classes: 36
Number of thread-cache bin size classes: 41
Number of large size classes: 196
Allocated: 94208, active: 102400, metadata: 13338640 (n_thp 0, edata 512, rtree 2097152), resident: 13357056, mapped: 21078016, retained: 1990656
Count of realloc(non-null-ptr, 0) calls: 0
Background threads: 0, num_runs: 0, run_interval: 0 ns
                           n_lock_ops (#/sec)       n_waiting (#/sec)      n_spin_acq (#/sec)  n_owner_switch (#/sec)   total_wait_ns   (#/sec)     max_wait_ns  max_n_thds
background_thread                   4       4               0       0               0       0               1       1               0         0               0           0
max_per_bg_thd                      0       0               0       0               0       0               0       0               0         0               0           0
ctl                                 2       2               0       0               0       0               1       1               0         0               0           0
prof                                0       0               0       0               0       0               0       0               0         0               0           0
prof_thds_data                      0       0               0       0               0       0               0       0               0         0               0           0
prof_dump                           0       0               0       0               0       0               0       0               0         0               0           0
prof_recent_alloc                   0       0               0       0               0       0               0       0               0         0               0           0
prof_recent_dump                    0       0               0       0               0       0               0       0               0         0               0           0
prof_stats                          0       0               0       0               0       0               0       0               0         0               0           0
arenas[0]:
name: "auto_0"
assigned threads: 0
uptime: 2000003
dss allocation precedence: "secondary"
decaying:  time       npages       sweeps     madvises       purged
   dirty: 10000            1            0            0            0
   muzzy:     0            0            0            0            0
                            allocated         nmalloc   (#/sec)         ndalloc   (#/sec)       nrequests   (#/sec)           nfill   (#/sec)          nflush   (#/sec)
small:                              0               1         1               1         1               1         1               0         0               0         0
large:                          94208               2         2               0         0               2         2               2         2               0         0
total:                          94208               3         3               1         1               3         3               2         2               0         0

active:                        102400
mapped:                      21078016
retained:                     1990656
base:                        13244432
internal:                       94208
metadata_edata:                   512
metadata_rtree:               2097152
metadata_thp:                       0
tcache_bytes:                       0
tcache_stashed_bytes:               0
resident:                    13357056
abandoned_vm:                       0
extent_avail:                       0
                           n_lock_ops (#/sec)       n_waiting (#/sec)      n_spin_acq (#/sec)  n_owner_switch (#/sec)   total_wait_ns   (#/sec)     max_wait_ns  max_n_thds
large                               2       2               0       0               0       0               1       1               0         0               0           0
extent_avail                        6       6               0       0               0       0               1       1               0         0               0           0
extents_dirty                       6       6               0       0               0       0               1       1               0         0               0           0
extents_muzzy                       2       2               0       0               0       0               1       1               0         0               0           0
extents_retained                    6       6               0       0               0       0               1       1               0         0               0           0
decay_dirty                         2       2               0       0               0       0               1       1               0         0               0           0
decay_muzzy                         2       2               0       0               0       0               1       1               0         0               0           0
base                              308     308               0       0               0       0               1       1               0         0               0           0
tcache_list                         4       4               0       0               0       0               1       1               0         0               0           0
hpa_shard                           0       0               0       0               0       0               0       0               0         0               0           0
hpa_shard_grow                      0       0               0       0               0       0               0       0               0         0               0           0
hpa_sec                             0       0               0       0               0       0               0       0               0         0               0           0
bins:           size ind    allocated      nmalloc (#/sec)      ndalloc (#/sec)    nrequests   (#/sec)  nshards      curregs     curslabs  nonfull_slabs regs pgs   util       nfills (#/sec)     nflushes (#/sec)       nslabs     nreslabs (#/sec)      n_lock_ops (#/sec)       n_waiting (#/sec)      n_spin_acq (#/sec)  n_owner_switch (#/sec)   total_wait_ns   (#/sec)     max_wait_ns  max_n_thds
                   8   0            0            1       1            1       1            1         1        1            0            0              0  512   1      1            0       0            0       0            1            0       0               5       5               0       0               0       0               1       1               0         0               0           0
                     ---
large:          size ind    allocated      nmalloc (#/sec)      ndalloc (#/sec)    nrequests (#/sec)  curlextents
                     ---
               28672  39        28672            1       1            0       0            1       1            1
                     ---
               65536  44        65536            1       1            0       0            1       1            1
                     ---
extents:        size ind       ndirty        dirty       nmuzzy        muzzy    nretained     retained       ntotal        total
                4096   0            1         4096            0            0            0            0            1         4096
                     ---
             2097152  31            0            0            0            0            1      1990656            1      1990656
                     ---
Bytes in small extent cache: 0
HPA shard stats:
  Purge passes: 0 (0 / sec)
  Purges: 0 (0 / sec)
  Hugeifies: 0 (0 / sec)
  Dehugifies: 0 (0 / sec)

  In full slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge
      ndirty: 0 huge, 0 nonhuge
      nretained: 0 huge, 0 nonhuge
  In empty slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge
      ndirty: 0 huge, 0 nonhuge
      nretained: 0 huge, 0 nonhuge
  In nonfull slabs:
                size ind npageslabs_huge    nactive_huge     ndirty_huge  npageslabs_nonhuge     nactive_nonhuge      ndirty_nonhuge   nretained_nonhuge
                     ---
--- End jemalloc statistics ---